consensus/safety-rules/src/tests/suite.rs

```diff
@@ -485,3 +486,89 @@ fn test_voting_potential_commit_id(func: RoundCallback) {
         a3.block().id(),
     );
 }
+
+fn test_sign_bad_proposals(func: RoundCallback) {
+    // The test constructs various malformed proposals for SafetyRule to sign
+
+    let (mut safety_rules, signer) = func();
+
+    let (proof, genesis_qc) = make_genesis::<Round>(&signer);
+    let round = genesis_qc.certified_block().round();
+    safety_rules.initialize(&proof).unwrap();
+
+    let a1 = test_utils::make_proposal_with_qc(round + 1, genesis_qc, &signer);
+    safety_rules.update(a1.block().quorum_cert()).unwrap();
+    let err = safety_rules.sign_proposal(a1.block().block_data().clone());
+    assert!(err.is_ok());
```

Comment:
recommend you just unwrap, that way if we get an err we can it is okay... also the variable name is a little strange since it is a result not an err or an ok